### PR TITLE
Housekeeping

### LINKED ISSUE #707: Manage test warnings: 'Agg backend'
Updates in our dependencies have created a lot of `PendingDeprecation` and `UserWarnings` in our tests, which we should address by ignoring them in the test spec or replacing the code that generates the warning with an appropriate expression that does not. Currently this involves the following actions:

- [x] ignore sklearn.exceptions.ConvergenceWarning in setup.cfg
- [x] replace `poof()` with `finalize()` in tests to remove matplotlib UserWarning about 'Agg' backend
- [x] document image comparison tests to reflect use of `poof()` and `finalize()` in tests
- [x] create issues to fix other warnings issued in tests

Note - warnings are generated in both the doc build and the tests; we should determine what all the unique warnings are and try to suppress them. 

See #705 for related discussion.

### LINKED ISSUE #1113: Housekeeping
- [x] rename master branch to main
- [x] create release issue template

@rebeccabilbro

### LINKED ISSUE #1127: Add .gitattributes to control line ending whitespace
Add a `.gitattributes` file to set per-repository defaults for line endings. This is in response to @lwgray's issue with whitespace changes when he moved to a windows machine. Depending on how `git` is configured it may checkout files with a `\r\n` (Windows) line ending rather than a `\n` line ending as on OS X and Linux. Setting a `.gitattributes` file will ensure that no matter our contributor's global git settings, that the files will be checked out correctly, preventing whitespace only commits. 

See: https://docs.github.com/en/free-pro-team@latest/github/using-git/configuring-git-to-handle-line-endings

Proposed `.gitattributes` file:

```
# Set the default behavior, in case contributors don't have core.autocrlf set.
* text=auto

# Baseline images are binary =and should not be modified
*.png binary
*.jpg binary
*.pdf binary
```

@lwgray is working on the solution on his machine which will allow us to figure out more clearly what is happening.

### PR DESCRIPTION
**Pull request by bbengfort**
This PR fixes #1113 by handling some housekeeping issues. We've changed our master branch to the main branch and created a release issue template (though I don't think it is used automatically by GitHub). We've also checked the new pip resolver and all of our dependencies are compatible. This PR also updates our CI to test Python 3.8 and 3.9 to ensure we're testing on the latest version of Python.

This PR also addresses some warnings that were generated by the tests and fixes #707 

This PR also adds a .gitattributes to fix #1127

Other housekeeping items:

- Apparently Sphinx `add_stylesheet` [is deprecated](https://github.com/sphinx-doc/sphinx/issues/7747) so updated to use `add_css_file` instead. 
- The cross-validation doc images were not building, so fixed the code by adding `shuffle=True` (scikit-learn deprecation)
- Adds a Sphinx build to GitHub ACtions for ensuring that the docs build correctly

### CODE REVIEW

File: docs/api/model_selection/cross_validation.rst [hunk 1]
```diff
@@ -41,7 +41,7 @@ In the following example, we show how to visualize cross-validated scores for a
     X, y = load_occupancy()
 
     # Create a cross-validation strategy
-    cv = StratifiedKFold(n_splits=12, random_state=42)
+    cv = StratifiedKFold(n_splits=12, shuffle=True, random_state=42)
```
Comment (bbengfort) [inline docs/api/model_selection/cross_validation.rst:44]: The docs were raising an exception because a random state was supplied without the shuffle argument set to true. The result was that the images were not shown in the build docs. In fact, they're not shown in our [current docs](https://www.scikit-yb.org/en/latest/api/model_selection/cross_validation.html#classification). Adding `shuffle=True` prevented the exception from being raised.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -3,104 +3,128 @@
...
-    strategy:
-      fail-fast: false
-      matrix:
-        python-version: [3.7, 3.8]
-        os: [ubuntu-latest, macos-latest, windows-latest]
-
-    runs-on: ${{ matrix.os }}
-
+  # Test building docs using Sphinx since this also runs a lot of Yellowbrick code
+  test_docs:
```
Comment (bbengfort) [inline .github/workflows/ci.yml:109]: Because of the docs build error that prevented the cross-validation visualizer to be shown in the docs I thought it might be good to have a GitHub action for testing the docs build. This test doesn't have to block a PR, but it will give us more visibility into the documentation.

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -3,104 +3,128 @@
...
       - develop
       - main
-      - master
+    tags:
+      - 'v*'
   # Trigger on pull request always (note the trailing colon)
   pull_request:
 
 jobs:
+  # Test Vanilla Python (using PyPI for dependencies) as a matrix.
```
Comment (bbengfort) [inline .github/workflows/ci.yml:18]: Most of the changes below are reordering to have the PyPI tests first then conda tests, as well as some whitespace and comments.

File: docs/conf.py [hunk 4]
```diff
@@ -232,7 +232,7 @@
 
 
 def setup(app):
-    app.add_stylesheet("theme_overrides.css")
+    app.add_css_file("theme_overrides.css")
```
Comment (bbengfort) [inline docs/conf.py:235]: Apparently, the Sphinx API has changed and this update was required … glad I build the docs!

File: docs/contributing/developing_visualizers.rst [hunk 5]
```diff
@@ -139,12 +139,14 @@ Visual tests are notoriously difficult to create --- how do you test a visualiza
             try:
                 visualizer = MyVisualizer()
                 assert visualizer.fit(X, y) is visualizer, "fit should return self"
-                visualizer.show()
+                visualizer.finalize()
```
Comment (bbengfort) [inline docs/contributing/developing_visualizers.rst:142]: Updates docs per #707

File: tests/test_classifier/test_prcurve.py [hunk 6]
```diff
@@ -435,7 +435,10 @@ def test_quick_method_with_test_set(self):
         )
 
         viz = precision_recall_curve(
-            RandomForestClassifier(random_state=72), X_train, y_train, X_test, y_test
+            RandomForestClassifier(random_state=72),
+            X_train, y_train,
+            X_test, y_test,
+            show=False
```
Comment (bbengfort) [inline tests/test_classifier/test_prcurve.py:441]: Ensure that quick method doesn't call `plt.show()` to prevent warnings per #707

File: tests/test_cluster/test_elbow.py [hunk 7]
```diff
@@ -432,7 +434,7 @@ def test_quick_method_params(self):
         custom_title = "My custom title"
         model = KMeans(3, random_state=13)
         oz = kelbow_visualizer(
-            model, X, sample_weight=np.ones(X.shape[0]), title=custom_title
+            model, X, sample_weight=np.ones(X.shape[0]), title=custom_title, show=False
```
Comment (bbengfort) [inline tests/test_cluster/test_elbow.py:437]: Ensure that quick method doesn't call `plt.show()` to prevent warnings per #707

File: tests/test_features/test_radviz.py [hunk 8]
```diff
@@ -138,7 +138,8 @@ def test_integrated_radviz_with_pandas(self):
 
         # Test the visualizer
         visualizer = RadViz()
-        visualizer.fit_transform_show(X, y)
+        visualizer.fit_transform(X, y)
+        visualizer.finalize()
```
Comment (bbengfort) [inline tests/test_features/test_radviz.py:142]: Ensure that quick method doesn't call `plt.show()` to prevent warnings per #707

File: .github/workflows/ci.yml (hunk 2 above)
Comment (rebeccabilbro) [reply .github/workflows/ci.yml:109]: Great idea &mdash; the docs behavior is always a little mysterious since it has it's own slightly different requirements.txt; this will be a great feature to add to our CI!

**Review by rebeccabilbro — APPROVED** [review]
Thank you so much @bbengfort for addressing these issues; it's been a while since we checked in on some of these things (new sklearn deprecations, sphinx, etc), and I'm surprised but happy that there weren't more things that needed to be addressed! 

I really like the idea of having GH Actions build the docs since the differences in local and hosted builds have  often caught me off guard. I'm noticing that GH Actions is flagging two lines in the rst as causing exceptions &mdash; both are inside plot directive plotting routines (line 99 in [learning curve](https://github.com/DistrictDataLabs/yellowbrick/blob/develop/docs/api/model_selection/learning_curve.rst#L99) and line 141 in [umap](https://github.com/DistrictDataLabs/yellowbrick/blob/develop/docs/api/text/umap_vis.rst#L141)). I wonder if this is related to the dependencies? Curious, but agree this should not block merging.

**Review by rebeccabilbro — APPROVED** [review]
